code/ExamplePlayer.cs

```diff
@@ -65,6 +65,28 @@ public override void Respawn()
         SupressPickupNotices = false;
     }
 
+    bool DesiresThirdPerson = false;
+
+    public void OnScopeStart()
+    {
+        if (!Game.IsServer) return;
+
+        if (CameraMode is ThirdPersonCamera)
+        {
+            CameraMode = new FirstPersonCamera();
+        }
+    }
+
+    public void OnScopeEnd()
+    {
+        if (!Game.IsServer) return;
+
+        if (DesiresThirdPerson)
+        {
+            CameraMode = new ThirdPersonCamera();
+        }
+    }
+
```

Comment:
Not a big fan of moving these out of the weapon base, could you explain your reasoning for this?

Reply:
If this isn't moved out of the weapon base then you are forcing consumers to use the `FirstPersonCamera` and `ThirdPersonCamera` classes which come with SWB. They might have a different camera system and don't want to use your camera. I think the weapon base should be as limited as possible to weapons. What if the game that's trying to use SWB is an isometric camera system, or third person only, or maybe a something entirely different?

Reply:
That's def something to revisit later on, as for the FOV modifiers they are already customizable and can be fully disabled.